velox/exec/RowContainer.h

```diff
@@ -255,7 +255,8 @@ class RowContainer {
   /// 'result' (starting at 'resultOffset') for the rows at positions in
   /// the 'rowNumbers' array. If a 'row' is null, sets
   /// corresponding row in 'result' to null. The positions in 'rowNumbers'
-  /// array can repeat and also appear out of order.
+  /// array can repeat and also appear out of order. If rowNumbers has a
```

Comment:
Using negative row number to request a null is a bit strange. There is already a way to add null, but have row number point to a null in the 'rows'. Is this not sufficient?

BTW, what's the use case you are trying to solve? Would be helpful to add that to the PR description.

Reply:
Regards pointing row number to a null in the 'rows',  it requires rows to have a null value already and the logic knowing its position.Or we could somehow artificially add a null row. This is complicated. 

We can imagine there are null rows surrounding rows, so a negative index position returns it (as per this logic).

In nth_value, if value in 'offset' row is null or it is out of bounds of [frameStart, frameEnd] then there is a null in the result.

Reply:
Could we not always have first (or last) element in 'rows' be nullptr? This way we have a fixed row number that produces a null.

Reply:
That solution does seem complex to me given the use-cases we have. The 'rows' behavior would be inconsistent between the various extractColumn APIs. In general in terms of reasoning its not that different to expect negative offsets to return nullptr vs 0 or last index returning nullptr I feel.

Reply:
We resolved to use negative rowNumbers based on nth_value code example.